test/metabase/query_processor/middleware/process_userland_query_test.clj

```diff
@@ -60,32 +64,33 @@
                                                       (respond metadata rows))]
                       (qp.pipeline/*run* query rff))))]
     (binding [driver/*driver* :h2]
-      (qp query qp.reducible/default-rff))))
+      (qp.store/with-metadata-provider (mt/id)
+        (qp query qp.reducible/default-rff)))))
 
 (deftest success-test
-  (let [query {:type ::success-test}]
+  (let [query {:database 2, :type :query, :query {:source-table 26}}]
```

Comment:
the tests were failing because this is an invalid mbql schema and converting legacy query to pmbql query is throwing an error.

I don't think it's possible for an invalid query to ever reach this place, so instead of guarding the implementation with try catch, I think it's better to update this test query.